cds_ils/importer/views.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# CDS-ILS is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+"""Views for importer."""
+
+import json
+import os
+import uuid
+from threading import Thread
+
+import arrow
+from flask import Blueprint, abort, copy_current_request_context, request
+from invenio_app_ils.permissions import need_permissions
+from webargs import fields
+from webargs.flaskparser import use_kwargs
+
+from cds_ils.importer.api import import_from_xml
+from cds_ils.importer.models import ImporterAgent, ImporterMode, \
+    ImporterTaskEntry, ImporterTaskLog
+
+
+def create_importer_blueprint(app):
+    """Add importer views to the blueprint."""
+    blueprint = Blueprint("invenio_app_ils_importer", __name__)
+
+    def allowed_files(filename):
+        """Checks the extension of the files."""
+        allowed_extensions = app.config[
+            "CDS_ILS_IMPORTER_FILE_EXTENSIONS_ALLOWED"
+        ]
+        return filename.lower().endswith(tuple(allowed_extensions))
+
+    def rename_file(filename):
+        """Renames filename with an unique name."""
+        unique_filename = uuid.uuid4().hex
+        ext = filename.rsplit(".", 1)[1]
+        return unique_filename + "." + ext
+
+    @blueprint.errorhandler(413)
+    def payload_too_large(error):
+        return {"message": "The file is too large", "status": 413}, 413
+
+    @blueprint.route("/importer/check/<int:log_id>", methods=["GET"])
```

Comment:
why not simply:
- `/importer/tasks/` for the list
- `/importer/tasks/<job_id>` for the details
(probably to be added to the new GH issue)